Simplified/Book/Models/NYPLBookRegistry.m

```diff
@@ -460,6 +467,16 @@ - (void)addBook:(NYPLBook *const)book
     @throw NSInvalidArgumentException;
   }
   
+  // Check to make sure audiobook bookmarks array contains NYPLAudiobookBookmark Object,
+  // otherwise, pass in an empty array
+  NSArray<NYPLAudiobookBookmark *> *newAudiobookBookmarks = [NSArray array];
+#if FEATURE_AUDIOBOOKS
+  if (audiobookBookmarks &&
```

Comment:
hopefully we can remove this check if we make the input array a typed array. You can definitely remove the `audiobookBookmarks &&` part since objc will safely fail the 2nd check if the array is nil.